<a href="https://colab.research.google.com/github/Alexander-1380/Yandex_practicum/blob/main/%D0%9F%D1%80%D0%B5%D0%B4%D0%BE%D0%B1%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D0%BA%D0%B0%20%D0%B4%D0%B0%D0%BD%D0%BD%D1%8B%D1%85%20-%20%D0%98%D1%81%D1%81%D0%BB%D0%B5%D0%B4%D0%BE%D0%B2%D0%B0%D0%BD%D0%B8%D0%B5_%D0%BD%D0%B0%D0%B4%D0%B5%D0%B6%D0%BD%D0%BE%D1%81%D1%82%D0%B8_%D0%B7%D0%B0%D0%B5%D0%BC%D1%89%D0%B8%D0%BA%D0%BE%D0%B2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Исследование надежности заемщиков


## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [ ]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [ ]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


Чтобы найти ответ на вопрос 3.5 в конце проекта, выведем занчения столбца total_income для строк в которых пропущено значение days_employed чтобы понять, в одних и тех же строках пропуски в данных или в разных </font>



In [ ]:
data.loc[data['days_employed'].isna()]['total_income'].unique()

array([nan])

результат - пропуски данных в столбцах total_income и days_employed в одних и тех же строках

## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [ ]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [ ]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [ ]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [ ]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [ ]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [ ]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [ ]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [ ]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [ ]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [ ]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [ ]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [ ]:
data.duplicated().sum()

71

In [ ]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [ ]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [ ]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

<ipython-input-19-0bc713f3049f>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['total_income_category'] = data['total_income'].apply(categorize_income)


**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [ ]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [ ]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [ ]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

<ipython-input-22-ecf9d79505dd>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['purpose_category'] = data['purpose'].apply(categorize_purpose)


### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [ ]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
data['debt'].unique() # посмотрим, что представлют из себя данные о задолженностях по возвратам кредитов.

array([0, 1])

значения в столбе "debt" принимают только два значения - 0 и 1, пропуски в столбце отсутствуют (выяснили это в Задании 12). Делаем вывод, что 1 означает, что клиент имел задолженности в своей истории, а 0 - что не имел.

Для каждого значения количества детей у клиентов посчитаем общее количество клиентов, а также количество клиентов,
имевших проблемы с возвратом кредита в срок. Вычислим долю проблемных от общего числа клиентов в процентах для
каждого количества детей

In [ ]:
# ИТЕРАЦИЯ_01
pivot_1 = data.pivot_table(index='children', values='debt', aggfunc=['sum','count'])
pivot_1['debt_rate_percent'] = pivot_1['sum'] / pivot_1['count'] * 100
pivot_1

sum  count debt_rate_percent
          debt   debt                  
children                               
0         1063  14091          7.543822
1          444   4808          9.234609
2          194   2052          9.454191
3           27    330          8.181818
4            4     41          9.756098
5            0      9          0.000000

Наблюдаем сильно отличающееся значение доли проблемных клиентов только для клинетов с 5 детьми.
Видим, что в данных всего 9 таких клиентов, на основе такой выборки сложно сделать замеры, считаем наблюдения для этой группы не значительными

**Вывод:** Согласно полученным данным, наименьшая вероятность не вернуть кредит в срок у клиентов без детей. Среди клиентов с детьми нельзя выделить какой-либо тренд в измненении доли проблемных клиентов в зависимости от количества детей. Заметно отличается доля для клиентов с 5-ю детьми, однако ей можно пренебречь из-за общего сравнительно малого количества таких клиентов.  Делаем вывод, что возврат кредита в срок зависит от наличия у клиента детей, но не зависит от их количества.

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [ ]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.

Из вывода основного датафрейма мы видим, что с данными о семейном положении клиентов связаны два столбца в датафрейме - family_status	и family_status_id
Установим связь одного с другим

In [ ]:
data.groupby('family_status_id')['family_status'].unique()

family_status_id
0          [женат / замужем]
1         [гражданский брак]
2           [вдовец / вдова]
3                [в разводе]
4    [Не женат / не замужем]
Name: family_status, dtype: object

Для хранения данных удобнее будет создать отдельную таблицу со столбцами family_status и family_status_id, а в основном датафрейме оставить только id, однако для решения текущего задания это действие не требуется.

Аналогично предыдущему заданию для каждого значения семейного положения посчитаем общее количество клиентов, а также количество клиентов, имевших проблемы с возвратом кредита в срок. Вычислим долю проблемных от общего числа клиентов в процентах для каждого значения семейного положения.

In [ ]:
pivot_2 = data.pivot_table(index='family_status', values='debt', aggfunc=['sum','count'])
pivot_2['debt_rate_percent'] = pivot_2['sum'] / pivot_2['count'] * 100
pivot_2.sort_values(by='debt_rate_percent', ascending=True)

sum  count debt_rate_percent
                      debt   debt                  
family_status                                      
вдовец / вдова          63    951          6.624606
в разводе               84   1189          7.064760
женат / замужем        927  12261          7.560558
гражданский брак       385   4134          9.313014
Не женат / не замужем  273   2796          9.763948

**Вывод:** наличие официально узаконенных отношений у клиента с партнером положительно влияет на вероятность возврата кредита в срок, причем даже в случае если эти отношения прекращены (развод) и смерти партнера (водвство). Разброс вероятностей невозврата кредита в срок для групп с узаконенными отношениями и групп без них в пределах (9.76 - 6.62) = 3,14%

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [ ]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.

Так как переменная уровня дохода является количественной, а не категориальной как в двух предыдущих заданиях, будем пользоваться уже выполненной нами в Задании 16 категоризацией клиентов по уровню дохода на группы:

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.

Эти данные содержатся в столбце 'total_income_category'.

Аналогично предыдущему заданию для каждой категории уровня дохода посчитаем общее количество клиентов, а также количество клиентов, имевших проблемы с возвратом кредита в срок. Вычислим долю проблемных от общего числа клиентов в процентах для каждой категории уровня дохода.


In [ ]:
pivot_3 = data.pivot_table(index='total_income_category', values='debt', aggfunc=['sum','count'])
pivot_3['debt_rate_percent'] = pivot_3['sum'] / pivot_3['count'] * 100
pivot_3.sort_values(by='debt_rate_percent', ascending=True)

sum  count debt_rate_percent
                       debt   debt                  
total_income_category                               
D                        21    349          6.017192
B                       354   5014          7.060231
A                         2     25          8.000000
C                      1353  15921          8.498210
E                         2     22          9.090909

Наблюдаем, что количество людей в категориях А и Е слишком мало для статистической значимости. Попробуем переопределить категорийность. Новая сетка категорий:
- 0–100000 — `'C'`;
- 100001–200000 — `'B'`;
- 200001 и выше — `'A'`.

In [ ]:
# переопределим функцию categorize_income()

def categorize_income(income):
    try:
        if 0 <= income <= 100000:
            return 'C'
        elif 100001 <= income <= 200000:
            return 'B'
        elif income > 200000:
            return 'A'
    except:
        pass

In [ ]:
# переопределим столбец 'total_income_category' и выведем для проверки уникальные значения столбца
data['total_income_category'] = data['total_income'].apply(categorize_income)
print(data['total_income_category'].unique())

['A' 'B' 'C']


In [ ]:
# создадим pivot_table для обновленного датафрейма
pivot_3 = data.pivot_table(index='total_income_category', values='debt', aggfunc=['sum','count'])
pivot_3['debt_rate_percent'] = pivot_3['sum'] / pivot_3['count'] * 100
pivot_3.sort_values(by='debt_rate_percent', ascending=True)

sum  count debt_rate_percent
                       debt   debt                  
total_income_category                               
A                       356   5039          7.064894
C                       354   4444          7.965797
B                      1022  11848          8.625928

Теперь видим, что категории сопоставимы по количеству людей, можем остановиться на этом.

**Вывод:** Наблюдаем, что наиболее надежными заёмщиками являются люди попадающие в категорию доходов от 200 000 руб и выше. Люди в категории доходов менее 100 000 руб менее надежные заемщики, а люди в категории доходов от 100 000 до 200 000 наименее надежные. нблюдается нелинейная зависимость вероятности возврата кредита в срок от уровня дохода. Разброс вероятности (8.63-7.06) = 1.57%

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [ ]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.

Воспользуемся уже выполненной нами в Задании 18 категоризацией клиентов по целям кредита:

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

Эти данные записаны в созданном в исходном датафрейме столбце 'purpose_category'.

Создадим пивот-таблицу pivot_4 по тому же алгоритму

In [ ]:
pivot_4 = data.pivot_table(index='purpose_category', values='debt', aggfunc=['sum','count'])
pivot_4['debt_rate_percent'] = pivot_4['sum'] / pivot_4['count'] * 100
pivot_4.sort_values(by='debt_rate_percent', ascending=True)

sum  count debt_rate_percent
                         debt   debt                  
purpose_category                                      
операции с недвижимостью  780  10751          7.255139
проведение свадьбы        183   2313          7.911803
получение образования     369   3988          9.252758
операции с автомобилем    400   4279          9.347978

**Вывод:** Наблюдаем, что наибольшая вероятность возврата кредита в срок при взятии его клиентом на проведение операций с недвижимостью, далее в порядке убывания вероятности следуют цели проведение свадьбы, получение образования и операции с автомобилем. Разброс вероятности (9.34-7.25) = 2.09%

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* пропуски в данных могли появиться из-за человеческого фактора - клиенты могли не заполнить соответствующие поля в электронных анкетах, а также из-за ошибки машинного считывания в случае заполнения клиентами анкет в бумажном виде

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* заполнять пропуски медианным значением оптимально так как:
- ячейку можно учитывать в арефмитических операциях с соответствующим столбцом (не будет возникать ошибки при попытки сложить значение формата None, Nan или строкового с числовым),
- медианное значение в ячейке не повлияет на общую картину данных по этому столбцу

### Шаг 4: общий вывод.

<div style="background: #B0E0E6; padding: 5px; border: 1px solid SteelBlue; border-radius: 5px;">
    <font color='4682B4'>
Общий вывод

Проведено исследование влияния на возврат в срок кредита четырех факторов:
- количества детей у клиента
- семейного положения клиента
- дохода клиента
- цели, на которую клиент берет кредит
Результаты исследования предполагается использовать в построении системы скорринга для оценки надежности новых клинетов, обращающихся в банк за кредитом.
Входные данные - статистика по клиентам.

Выходные данные - доля клиентов, имевших историю не возврата кредитов в срок от общего числа клиентов для каждой категории исследуемых параметров.

- Гипотеза 1 - количество детей у клиента влияет на возврат им кредита в срок
- Гипотеза 2 - семейное положение клиента влияет на возврат им кредита в срок
- Гипотеза 3 - уровень дохода клиента влияет на возврат им кредита в срок
- Гипотеза 4 - цель кредита клиента влияет на возврат им кредита в срок

После проверки удалось установить следующее:


**Гипотеза 1** подтверждена частично.

Результаты исследования:
* 7.54% - доля не возвращенных в срок кредитов для клиентов без детей
* 9.23% - доля не возвращенных в срок кредитов для клиентов с 1 ребенком
* 9.45% - доля не возвращенных в срок кредитов для клиентов с 2 детьми
* 8.18% - доля не возвращенных в срок кредитов для клиентов с 3 детьми

Выводы: наиболее надежными заемщиками являются клиенты без детей. Доля не возвращенных в срок кредитов увеличивается для клиентов с 1 и 2 детьми.

Рекомендация: для более точного исследования клинетов с количеством детей от 3 и больше необходимо предоставить статистику с бОльшим количеством таких клиентов.


**Гипотеза 2** подтверждена.

Результаты исследования:
* 6.56% - доля не возвращенных в срок кредитов для клиентов со статусом вдовец / вдова
* 7.11% - доля не возвращенных в срок кредитов для клиентов со статусом в разводе
* 7.52% - доля не возвращенных в срок кредитов для клиентов со статусом женат / замужем
* 9.29% - доля не возвращенных в срок кредитов для клиентов со статусом гражданский брак
* 9.74% - доля не возвращенных в срок кредитов для клиентов со статусом Не женат / не замужем

Выводы: Наличие официально узаконенных отношений у клиента с партнером положительно влияет на вероятность возврата кредита в срок, причем даже в случае, если эти отношения прекращены (развод) и смерти партнера (водвство).


**Гипотеза 3** подтверждена.

Результаты исследования:
* 7.93% - доля не возвращенных в срок кредитов для клиентов с уровнем дохода от 0 до 100 000 руб
* 8.74% - доля не возвращенных в срок кредитов для клиентов с уровнем дохода от 100 000 до 200 000 руб
* 7.07% - доля не возвращенных в срок кредитов для клиентов с уровнем дохода от 200 000 руб и выше

Выводы: Наиболее надежные заемщики клиенты с доходом от 200 000 руб, дальше следует группа людей с доходом ниже 100 000 руб, и наименее надежные клиенты с уровнем дохода от 100 000 до 200 000 руб

Рекомендации: для более точного исследования клинетов и разделения их на бОльшее число категорий, необходимо предоставить статистику с бОльшим количеством клиентов с уровнем доходов от 0 до 50 000 руб и от 200 000 руб и выше.


**Гипотеза 4** подтверждена.

Результаты исследования:
* 7.21% - доля не возвращенных в срок кредитов взятых клиентом на операции с недвижимостью
* 7.92% - доля не возвращенных в срок кредитов взятых клиентом на проведение свадьбы
* 9.20% - доля не возвращенных в срок кредитов взятых клиентом на получение образования
* 9.34% - доля не возвращенных в срок кредитов взятых клиентом на операции с автомобилем

Выводы: Наименее ответственными заемщиками являются люди, берущие кредит на покупку автомобиля и получения образования.


**Общие выводы:** все 4 исследованных фактора влияют на платежеспособность замещиков. При проектировании системы скорринга для оценки платежеспособности клиентов рекомендуется присваивать клиентам из соответствующих категорий весовые коэффициенты пропорционально долям не возвращенных в срок кредитов для каждой исследованной категории.</font>
</div>